File: 27-Remove-Element.c
int removeElement(int* nums, int numsSize, int val) {
    int c=0;
    if(numsSize == 0)
        return 0;
    int j=numsSize-1;
    while(nums[j]==val){
        j--;
        if(j<0){
            return 0;
        }
    }
    for(int i=0;i<numsSize;i++){
        if(val == nums[i])
            c++;
    }
    for(int i=0;i<numsSize-c ;i++){
        if(val==nums[i]){
            int t = nums[i];
            nums[i] = nums[j];
            nums[j] = t;
            j--;
        }
        while(nums[j]==val){
        j--;
        if(j<0){
            return 0;
        }
    }  
    }
    return numsSize - c;
}

File: 27-Remove-Element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int n = nums.length;
        if(n == 0)
            return 0;
        int left = 0;
        int right = n - 1;
        while(left < right){
            while(left < n && nums[left] != val)
                left++;
            while(right >= 0 && nums[right] == val)
                right--;
            if(left < right){
                swap(nums,left,right);
                left++;
                right--;
            }
        }
        return left;
    }
    public void swap(int [] nums,int i,int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}